src/components/structures/RoomView.js

```diff
@@ -191,18 +191,32 @@ module.exports = React.createClass({
             newState.room = MatrixClientPeg.get().getRoom(newState.roomId);
         }
 
+        if (this.state.roomId !== newState.roomId) {
+            // Store the scroll state for the previous room so that we can return to this
+            // position when viewing this room in future.
+            if (this.state.roomId) {
+                RoomScrollStateStore.setScrollState(this.state.roomId, this._getScrollState());
```

Comment:
... and this is redundant. We save the scrollState in unmount.

Reply:
oh, as per above